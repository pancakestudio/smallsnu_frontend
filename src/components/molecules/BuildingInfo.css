.resList {
  border: 0px !important;
  margin: 10px 0px;
}

.postList {
  border: 0px !important;
  margin: 10px 0px;
}

.semiList {
  border: 0px !important;
  margin: 10px 0px;
}
