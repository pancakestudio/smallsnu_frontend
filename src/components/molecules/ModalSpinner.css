.spinnerWrapper {
  text-align: center;  
}

.spinnerWrapper .spinner {
  width: 100px;
  height: 100px;
}
