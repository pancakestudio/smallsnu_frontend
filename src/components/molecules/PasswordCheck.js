import React from 'react';
import { Modal, Button, Form } from 'react-bootstrap'
import './PasswordCheck.css'

export const PasswordCheck = ({show, target, onHideModal, onDeletePost, onDeleteComment}) => {
  let passwordInput
  const handleSubmit = (e) => {
    e.preventDefault()
    const password = passwordInput.value
    if("building" in target){
      const bldgNo = target.building.code
      onDeletePost(target, bldgNo, password)
    } else {
      const postId = target.post
      onDeleteComment(target, postId, password)
    }
  }

  return(
    <Modal
      show = {show}
      onHide = {onHideModal}
      backdrop = {false}
      size = "sm"
      dialogClassName = "passwordCheck"
      centered
    >
      <Modal.Header closeButton>
        <Modal.Title>비밀번호 확인</Modal.Title>
      </Modal.Header>
      <Form onSubmit={handleSubmit}>
        <Modal.Body>
          <Form.Group controlId="formGridPassword">
            <Form.Control type="password" placeholder="비밀번호" ref = {(ref) => (passwordInput=ref)}/>
          </Form.Group>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="danger" type="submit" onClick = {handleSubmit}>삭제</Button>
        </Modal.Footer>
      </Form>
    </Modal>
  )
}
