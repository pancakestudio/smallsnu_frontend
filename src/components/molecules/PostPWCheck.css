.PostPwCheck {
  min-width: 30vw;
}
