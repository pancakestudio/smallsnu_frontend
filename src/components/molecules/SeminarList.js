import React from 'react'
import { Card } from 'react-bootstrap'
import { SemiPreview } from '../atoms/SemiPreview'
import { historyPush } from '../../utils/Functions'
import './SeminarList.css'

export const SeminarList = ({semis, page, onSeminarClick}) => {
  const handleSeminarClick = (id) => {
    historyPush(`/seminar/${id}`)
  }
  let start, end

  start = (page-1)*3
  end = start+3 < semis.length ? (start+3) : (semis.length)
  return (
    <Card className="seminarList border-0">
      {semis.slice(start, end).map((semi) => (
        <SemiPreview key={semi.id} {...semi} onClick={()=>{handleSeminarClick(semi.id)}} />
      ))}
    </Card>
  )
}
