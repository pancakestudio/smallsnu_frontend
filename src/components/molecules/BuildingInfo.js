import React from 'react'
import { Card, Tabs, Tab, Button } from 'react-bootstrap'
import { AmenityPreview } from '../atoms/AmenityPreview'
import { PostPreview } from '../atoms/PostPreview'
import { SemiPreview } from '../atoms/SemiPreview'
import { historyPush } from '../../utils/Functions'
import './BuildingInfo.css'

export const BuildingInfo = ({bldgNo, rests, semis, posts}) => {
  const handlePostClick = (post) => {
    historyPush(`/post/${post.id}`)
  }

  const handleBoardClick = () => {
    historyPush(`/board/${bldgNo}`)
  }

  const handleSeminarClick = (semi) => {
    historyPush(`/seminar/${semi.id}`)
  }

  const handleSeminarListClick = () => {
    historyPush(`/seminarlist/${bldgNo}`)
  }

  const handleResClick = (res) => {
    historyPush(`/restaurant/${res.id}`)
  }
  let resList, postList, semiList

  // Restaurant List
  if(rests && rests.length!==0){
    resList = (
      <Card className="resList">
        {rests.map((rest) => (
          <AmenityPreview key={rest.id} {...rest} onClick={()=>{handleResClick(rest)}}/>
        ))}
      </Card>
    )
  } else {
    resList = (
      <Card className="resList">
        <Card.Body>
          <Card.Text>식당 정보가 없습니다.</Card.Text>
        </Card.Body>
      </Card>
    )
  }

  // Post List
  if(posts && posts.length!==0){
    postList = (
      <Card className="postList">
        {posts.slice(0, 3).map((post) => (
          <PostPreview key={post.id} {...post} onClick={()=>handlePostClick(post)}/>
        ))}
        <Card.Footer> <Button className="postListButton" variant="link" onClick={()=>{handleBoardClick()}}>게시글 더보기</Button> </Card.Footer>
      </Card>
    )
  } else {
    postList = (
      <Card className="postList">
        <Card.Body>
          <Card.Text>게시판에 글이 없습니다.</Card.Text>
        </Card.Body>
        <Card.Footer> <Button className="postListButton" variant="link" onClick={()=>{handleBoardClick()}}>게시판 바로가기</Button> </Card.Footer>
      </Card>
    )
  }

  // Seminar List
  if(semis && semis.length!==0){
    semiList = (
      <Card className="semiList">
        {semis.slice(0, 2).map((semi) => (
          <SemiPreview key={semi.id} {...semi} onClick={()=>handleSeminarClick(semi)}/>
        ))}
        <Card.Footer> <Button className="semiListButton" variant="link" onClick={()=>{handleSeminarListClick()}}>세미나 더보기</Button> </Card.Footer>
      </Card>
    )
  } else {
    semiList = (
      <Card className="semiList">
        <Card.Body>
          <Card.Text>세미나 정보가 없습니다.</Card.Text>
        </Card.Body>
      </Card>
    )
  }

  return(
    <Tabs className="buildingResList">
      <Tab eventKey="restaurant" title="식당 정보">
        { resList }
      </Tab>
      <Tab eventKey="seminar" title="세미나 정보">
        { semiList }
      </Tab>
      <Tab eventKey="post" title="건물 게시판">
        { postList }
      </Tab>
    </Tabs>
  )
}
