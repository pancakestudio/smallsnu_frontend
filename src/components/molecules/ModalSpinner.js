import React from 'react'
import { Modal, Spinner } from 'react-bootstrap'
import './ModalSpinner.css'

export const ModalSpinner = () => {
  return (
    <Modal
      show = {true}
      dialogClassName = "loadingModal"
      centered
    >
      <div className="spinnerWrapper">
        <Spinner className="spinner" animation="grow" variant="light"/>
      </div>
    </Modal>
  )
}
