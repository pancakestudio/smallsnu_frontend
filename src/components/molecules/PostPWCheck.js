import React from 'react';
import { Modal, Button, Form } from 'react-bootstrap'
import { FaAngleLeft } from 'react-icons/fa'
import { historyPush } from '../../utils/Functions'
import './PostPWCheck.css'

export const PostPWCheck = ({show, post, bldgNo, onHideModal, onDelete}) => {
  let passwordInput
  const handleSubmit = () => {
    if(post.password === passwordInput.value){
      onDelete(post, bldgNo)
      onHideModal()
    }else{
      alert('비밀번호가 틀렸습니다')
    }
  }
  const handleBack = () => {
    onHideModal()
  }

  return(
    <Modal
      show = {show}
      onHide = {() =>{historyPush('/')}}
      dialogClassName = "PostPWCheck"
      centered
    >
      <Modal.Header closeButton>
      <Button className="back" onClick={handleBack}><FaAngleLeft /></Button>
        <Modal.Title>비밀번호 확인</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Form.Group controlId="formGridPassword">
          <Form.Control type="password" placeholder="비밀번호" ref = {(ref) => (passwordInput=ref)}/>
        </Form.Group>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="primary" onClick = {handleSubmit}>확인</Button>
      </Modal.Footer>
    </Modal>
  )
}
