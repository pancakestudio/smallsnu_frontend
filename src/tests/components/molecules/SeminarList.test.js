import React from 'react'
import { SeminarList } from '../../../components/molecules/SeminarList'
import { shallow } from 'enzyme'

describe('SeminarList', ()=>{
  let component
  const seminars = [
    {
      "id": 1,
      "title": "[Seminar] Autonomous Driving: Simulation and Navigation",
      "talker": "Dinesh ManochaDepartment of Computer Science and Electrical & Computer Engineering, University of Maryland at College Park",
      "where": "302동 309-1호",
      "time": "2019년 3월 28일 목요일 PM 1:30 - 2019년 3월 28일 목요일 PM 2:30",
      "link": "https://cse.snu.ac.kr/node/36950"
    },
    {
      "id": 2,
      "title": "[Seminar] Augmented Virtuality with 360 Videos",
       "talker": "Taehyun Rhee교수Victoria University of Welington",
       "where": "302동 308호",
       "time": "2019년 3월 22일 금요일 PM 1:30 - 2019년 3월 22일 금요일 PM 2:45",
       "link": "https://cse.snu.ac.kr/node/36887"
    },
    {
      "id": 11,
      "title": "[Seminar] SCAI AI Lunch Talk #1 - Deep Learning @ Naver/Line",
      "talker": "김성훈리더NAVER Clova AI ",
      "where": "302동 105호",
      "time": "2019년 3월 21일 목요일 PM 12:30 - 2019년 3월 21일 목요일 PM 1:30",
      "link": "https://cse.snu.ac.kr/node/36870"
    },
    {
      "id": 12,
      "title": "[Seminar] Scalable and Automatic Vulnerability Discovery Beyond Random Testing",
      "talker": "윤인수 (Insu Yun)Ph.D. student Georgia Institute of Technology",
      "where": "302-409",
      "time": "2019년 3월 11일 월요일 AM 10:30 - 2019년 3월 11일 월요일 PM 12:00",
      "link": "https://cse.snu.ac.kr/node/35622"
    },
  ]
  const page = 1
  const mockSeminarClick = jest.fn()

  it('renders correctly', ()=>{
    component = shallow(
      <SeminarList
        semis={seminars.slice(0, 2)}
        page={page}
      />
    )
  })

  it('matches snapshot', ()=>{
    expect(component).toMatchSnapshot()
  })

  it('has semipreviews', ()=>{
    expect(component.find('SemiPreview').exists()).toBe(true)
    expect(component.find('SemiPreview').length).toBe(2)
  })

  it('deals with a long seminar list', ()=>{
    component.setProps({semis: seminars})
    expect(component.find('SemiPreview').exists()).toBe(true)
    expect(component.find('SemiPreview').length).toBe(3)
  })

  it('calls functions', ()=>{
    component.find('SemiPreview').at(0).prop('onClick')()
    expect(global.window.location.pathname).toEqual('/seminar/1')
  })
})
