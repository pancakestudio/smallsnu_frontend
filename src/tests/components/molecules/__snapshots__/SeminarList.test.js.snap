// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeminarList matches snapshot 1`] = `ShallowWrapper {}`;
